v3/util/setup-credentials.sh

```diff
@@ -6,36 +6,60 @@ OWNER=$(whoami)
 
 source /etc/environment
 
-if [ ! -z $SECURE_FILES ]; then
-    # have to use  us-east-1 - aws tool does not recognize anything else
-    # for S3 download
-    AWS_CREDS=""
-    if [ ! -z $AWS_ACCESS_KEY ]; then
-        AWS_CREDS=" -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY \
-         -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_KEY "
+TABLE=`sudo echo $SECRETS_TABLE`
+docker pull behance/docker-aws-secrets-downloader:latest
+
+# Get all available secrets and configs
+AV_SECRETS=`sudo docker run behance/docker-aws-secrets-downloader --table $TABLE --key secrets`
+AV_CONFIGS=`sudo docker run behance/docker-aws-secrets-downloader --table $TABLE --key configs`
+
+printf "#!/bin/bash\nsource /etc/environment\n" >  ${HOMEDIR}/.etcd_loader.sh
+
+echo "$AV_SECRETS" | while read line ; do
+    SECRET=`sudo docker run behance/docker-aws-secrets-downloader --table $TABLE --key secrets --name $line`
+    SECRET_TYPE=`echo $SECRET | cut -d' ' -f2`
+    SECRET_PATH=`echo $SECRET | cut -d' ' -f3`
+    SECRET_VAL=`echo $SECRET | cut -d' ' -f4-`
+
+    if [[ "$SECRET_TYPE" = "etcd" ]]; then
+        echo "Setting ETCD value for secret: $SECRET_PATH"
+        #etcdctl set $SECRET_PATH $SECRET_VAL &>/dev/null
+        printf "etcdctl set %s %s &>/dev/null \n" "$SECRET_PATH" "$SECRET_VAL" >> ${HOMEDIR}/.etcd_loader.sh
     fi
-    sudo docker run --rm \
-        -v ${HOMEDIR}:/data/  $AWS_CREDS behance/docker-aws-s3-downloader \
-         us-east-1 $CONTROL_TIER_S3SECURE_BUCKET $SECURE_FILES
-
-    # must chown all files to core for use
-    FILES=(${SECURE_FILES//:/ })
-    for file in "${FILES[@]}"; do
-        PIECES=(${file//,/ })
-        S3FILE=${PIECES[0]}
-        PERMS=${PIECES[1]}
-        TARGET=${PIECES[2]}
-
-        if [ "${TARGET}" == "" ]; then
-            TARGET="${S3FILE}"
-        fi
-
-        sudo chown -R ${OWNER}:${OWNER} ${HOMEDIR}/${TARGET}
-    done
+done
+
+echo "$AV_CONFIGS" | while read line ; do
+    CONFIG=`docker run behance/docker-aws-secrets-downloader --table $TABLE --key configs --name $line`
+    CONFIG_PATH=`echo $CONFIG | cut -d' ' -f1`
+    CONFIG_VAL=`echo $CONFIG | cut -d' ' -f2-`
+
+    echo "Setting ETCD value for config: $CONFIG_PATH"
+    #etcdctl set $CONFIG_PATH $CONFIG_VAL &>/dev/null
+    printf "etcdctl set %s %s &>/dev/null \n" "$CONFIG_PATH" "$CONFIG_VAL" >> ${HOMEDIR}/.etcd_loader.sh
+done
+
+chmod +x ${HOMEDIR}/.etcd_loader.sh
+${HOMEDIR}/.etcd_loader.sh
+rm ${HOMEDIR}/.etcd_loader.sh
+
+# Create a dockercfg
+DOCKERCFG_CONTENTS=`sudo docker run behance/docker-aws-secrets-downloader --table $TABLE --key secrets --name DOCKERCFG --format plain`
```

Comment:
It's explicit because there is custom work done around it (saving it to a file, copying to root). In v2, before this update, there was also custom work done for .dockercfg.

Also, generic secrets just get loaded into etcd. I didn't want to put the contents of a file into etcd.